src/engine/BMGame.php

```diff
@@ -476,10 +380,13 @@ protected function do_next_step_choose_join_game() {
     }
 
     protected function update_game_state_choose_join_game() {
-        if (isset($this->hasPlayerAcceptedGameArray) &&
-            is_array($this->hasPlayerAcceptedGameArray) &&
-            in_array(FALSE, $this->hasPlayerAcceptedGameArray)) {
-            foreach ($this->hasPlayerAcceptedGameArray as $playerIdx => $hasAccepted) {
+        $hasPlayerAcceptedGameArray = array_fill(0, $this->nPlayers, NULL);
+        foreach ($this->playerArray as $playerIdx => $player) {
+            $hasPlayerAcceptedGameArray[$playerIdx] = $player->hasPlayerAcceptedGame;
+        }
+
+        if (in_array(FALSE, $hasPlayerAcceptedGameArray)) {
+            foreach ($hasPlayerAcceptedGameArray as $playerIdx => $hasAccepted) {
```

Comment:
Worth noting: i'm being intentionally picky here to see if you agree with my thinking about this.